components/org.wso2.carbon.identity.application.authenticator.facebook/src/main/java/org/wso2/carbon/identity/application/authenticator/facebook/FacebookAuthenticator.java

```diff
@@ -181,6 +207,19 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
                             .setScope(scope).setState(state)
                             .buildQueryMessage();
             response.sendRedirect(authzRequest.getLocationUri());
+            if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                DiagnosticLog.DiagnosticLogBuilder diagnosticLogBuilder = new DiagnosticLog.DiagnosticLogBuilder(
+                        OUTBOUND_AUTH_FACEBOOK_SERVICE, INITIATE_OUTBOUND_AUTH_REQUEST);
+                diagnosticLogBuilder.logDetailLevel(DiagnosticLog.LogDetailLevel.APPLICATION)
+                        .resultStatus(DiagnosticLog.ResultStatus.SUCCESS)
+                        .inputParam(LogConstants.InputKeys.STEP, context.getCurrentStep())
+                        .inputParam(LogConstants.InputKeys.IDP, context.getExternalIdP().getIdPName())
+                        .inputParam("authenticator properties", authenticatorProperties.keySet())
+                        .inputParam(LogConstants.InputKeys.SCOPE, scope)
+                        .inputParams(getApplicationDetails(context))
+                        .resultMessage("Redirecting to the Facebook login page.");
```

Comment:
can we keep the same order of methods?

Reply:
Fixed with c4f5d3aa5cbb1291e581f242cc9aba81ef9f19ac